File: fall/image_processing/homework/hw1/main.py
import numpy as np
import cv2
import matplotlib.pyplot as plt
from argparse import ArgumentParser
from image_processing import ImageProcessor
from PIL import Image

filename = 'p1im2.png'

def main():

    # Open the image
    imgProc = ImageProcessor()
    image = imgProc.process(filename)

    Histogram = imgProc.histogram(image)


    
    
    tmp = [i for i in range(Histogram.shape[0])] 
   
    normalized = imgProc.normalize_histogram(Histogram)
    HSL = cv2.cvtColor(image, cv2.COLOR_BGR2HSV)
    plt.imshow(HSL)
    #cv2.imshow(HSL)
    
    #plt.bar(tmp, normalized[:,0], color='red')
    #plt.bar(tmp, normalized[:,1], color='green')
    #plt.bar(tmp, normalized[:,2], color='blue')
    
    #RGB = imgProc.HSL_to_RGB(HSL)
    print(RGB)
    #plt.imshow(RGB)
    plt.show()
    
    


if __name__=='__main__':
    main()

File: fall/image_processing/homework/hw1/main.cpp
#include<iostream>

#include<cstdio>
#include<string>
#include "image_processing.h"

int main(int argc, char* argv[]){

	if(argc == 1){
		printf("Please specify an image filename.\n");	
		return 1;
	}
	ImageProcessor imgProc;
	std::string filename = argv[1];
	imgProc.process(filename, "contrast adjustment");	
	return 0;
}
